cdc/sink/dmlsink/mq/dispatcher/event_router.go

```diff
@@ -267,7 +269,12 @@ func getTopicDispatcher(
 			return nil, cerror.WrapError(cerror.ErrKafkaInvalidConfig, err)
 		}
 
-		if p == config.ProtocolAvro {
+		if schema == sink.PulsarScheme {
```

Comment:
`If` and `Else if` is not checking the same element, so I would prefer rewrite this logic.

`If` is checking about the `schem`, `Else if` is checking about the `protocol`.